File: src/main/java/com/dansfarmaroots/blog/blog/Blog.java
package com.dansfarmaroots.blog.blog;

import com.dansfarmaroots.blog.user.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "blogs")
public class Blog {
    // Getters and Setters
    @Id
    private String id;
    private String title;
    private String content;
    private Date publishDate;
    private String authorName;
}

File: src/lib/models/Blog.ts
interface IBlog{
    id: string | null;
    title: string;
    content: string;
    publishDate: string;
}

export class BlogCreation implements IBlog{
    id: null;
    title: string;
    content: string;
    publishDate: string;
    authorName: string;


    constructor(title: string, content: string, publishDate: string, authorName: string) {
        this.id = null;
        this.title = title;
        this.content = content;
        this.publishDate = publishDate;
        this.authorName = authorName;
    }
}

export class Blog implements IBlog{
    id: string;
    title: string;
    content: string;
    publishDate: string;
    authorName: string;


    constructor(id: string, title: string, content: string, publishDate: string, authorName: string) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.publishDate = publishDate;
        this.authorName = authorName;
    }
}
